netbox/ipam/forms/model_forms.py

```diff
@@ -369,6 +369,19 @@ def clean(self):
                 'primary_for_parent', _("Only IP addresses assigned to an interface can be designated as primary IPs.")
             )
 
+        if self.instance.pk:
+            prev_interface = self.instance.interface.first() or self.instance.vminterface.first()
+            # If the prev interface exists and does not match the new interface, we need to validate it isn't set as primary
+            # for the parent
+            if prev_interface and prev_interface != interface:
+                prev_parent = prev_interface.device or prev_interface.virtual_machine
+                # Check that the parent exists and if it is set as a primary ip.
+                if prev_parent and prev_parent.primary_ip4 == self.instance or prev_parent.primary_ip6 == self.instance:
+                    self.add_error(
+                        selected_objects[0],
+                        _("Cannot reassign IP address while it is designated as the primary IP for the parent object")
+                    )
+
```

Comment:
Why is this needed? Decoupling the ValidationError from a specific field under the model's `clean()` seems sufficient to address the bug.

Reply:
It likely isn't needed, but it was an attempt to maintain the field highlighting.  However you are correct that it likely isn't needed with just the decoupling.